modules/sdp/SDP.js

```diff
@@ -462,32 +472,91 @@ export default class SDP {
                 ssrc = assrcline.substring(7).split(' ')[0];
             }
 
+            const contents = $(elem.tree()).find(`content[name='${mediaType}']`);
+
+            // Append source groups from the new m-lines to the existing media description. The SDP will have multiple
+            // m-lines for audio and video including the recv-only ones for remote sources but there needs to be only
+            // one media description for a given media type that should include all the sources, i.e., both the camera
+            // and screenshare sources should be added to the 'video' description.
+            for (const content of contents) {
+                if (!content.hasAttribute('creator')) {
+                    // eslint-disable-next-line no-continue
+                    continue;
+                }
+
+                if (ssrc) {
+                    const description = $(content).find('description');
+                    const ssrcMap = SDPUtil.parseSSRC(mediaItem);
+
+                    for (const [ availableSsrc, ssrcParameters ] of ssrcMap) {
+                        const sourceName = SDPUtil.parseSourceNameLine(ssrcParameters);
+                        const videoType = SDPUtil.parseVideoTypeLine(ssrcParameters);
+                        const source = Strophe.xmlElement('source', {
+                            ssrc: availableSsrc,
+                            name: sourceName,
+                            videoType,
+                            xmlns: XEP.SOURCE_ATTRIBUTES
+                        });
+
+                        const msid = SDPUtil.parseMSIDAttribute(ssrcParameters);
+
+                        if (msid) {
+                            const param = Strophe.xmlElement('parameter', {
+                                name: 'msid',
+                                value: msid
+                            });
+
+                            source.append(param);
+                        }
+                        description.append(source);
+                    }
+
+                    const ssrcGroupLines = SDPUtil.findLines(mediaItem, 'a=ssrc-group:');
+
+                    ssrcGroupLines.forEach(line => {
+                        const idx = line.indexOf(' ');
+                        const semantics = line.substr(0, idx).substr(13);
```

Comment:
Where do these 13 and 14 numbers come from?

Reply:
length of 'a=ssrc-group:'

Reply:
I wish there was a better way...

Reply:
I added a helper function in SDPUtil for this.